Add file name and line number info to log messages

lawrence-forooghian opened the pull request:
This adds a tag like `@(MyFile.swift:130)` to each log message, identifying the source code file and line number from which the log message was emitted.

Please review commit-by-commit (it has a couple of refactors that might be difficult to review otherwise) and see commit messages for more details.

Closes #494.

QuintinWillison reviewed (approved):
I don't understand everything I'm looking at due to lack of Swift knowledge, plus lack of depth of knowledge in this SDK, however this change seems very sensible and has got to make things better with nothing looking obviously out of place to me.

I have a couple of thoughts which are perhaps outside scope of this pull request, but might be worth considering:

- Would it be worthwhile adding some guidance to [the contributing guide](https://github.com/ably/ably-asset-tracking-swift/blob/main/CONTRIBUTING.md) relating to adding log output to this SDK?
- Does this change need to include modification to the [Generating mocks](https://github.com/ably/ably-asset-tracking-swift/blob/main/CONTRIBUTING.md#generating-mocks) section of the aforementioned contributing guide?

lawrence-forooghian commented:
> * Would it be worthwhile adding some guidance to [the contributing guide](https://github.com/ably/ably-asset-tracking-swift/blob/main/CONTRIBUTING.md) relating to adding log output to this SDK?

Good point. It's probably not directly relevant to this PR; I should have done it in #508. But I've added 848d1dfb7d0277720acb3828df2e99a64319ac4b to address it here.

> * Does this change need to include modification to the [Generating mocks](https://github.com/ably/ably-asset-tracking-swift/blob/main/CONTRIBUTING.md#generating-mocks) section of the aforementioned contributing guide?

It does, thanks! Added 539d5e030c9ae4da7b2cda5fa166dd8c630b2841.
